src/Components/Web.JS/src/Services/NavigationManager.ts

```diff
@@ -80,7 +80,11 @@ export function navigateTo(uri: string, forceLoad: boolean, replace: boolean = f
     history.replaceState(null, '', temporaryUri);
     location.replace(uri);
   } else if (replace){
-    history.replaceState(null, '', absoluteUri)
+      if (forceLoad) {
+        location.replace(uri);
+      } else {
+        history.replaceState(null, '', absoluteUri)
```

Comment:
This logic looks wrong to me. What happens if `isWithinBaseUriSpace(absoluteUri)` is false, `force` is false, and `replace` is true?

What I'd expect is: since this is an external navigation, the force param is irrelevant, and the `replace` flag means we should be calling `location.replace`.

What the logic appears to do is call `history.replaceState`, which won't produce a useful behavior because it's only doing HTML5-style client-side navigation and won't trigger an actual external page load.

Here's the full truth table I'd expect:

| isWithinBaseUriSpace(absoluteUri) | Force | Replace | Expected behavior |
| ------------- |-------------| -----| ---| 
| false | false | false | `location.href = url` (append external page load to history) |
| false | false | true | `location.replace` (replace with external page load on history) |
| false | true | false | `location.href = url` (append external page load to history) |
| false | true | true | `location.replace` (replace with external page load on history) |
| true | false | false | `performInternalNavigation(absoluteUri, false, false)` (calls `history.pushState`) |
| true | false | true | `performInternalNavigation(absoluteUri, false, true)` (calls `history.replaceState`) |
| true | true | false | `location.href = url` (append external page load to history) |
| true | true | true | `location.replace` (replace with external page load on history) |

Do you agree with this list? Could you check how it corresponds to the implementation? I think the only combination that's currently wrong is the one I mentioned at the top of this comment.

I suspect that before this PR, the logic here was already wrong in that it didn't handle `replace` without also `force`. That was OK before because it wasn't really proper public API. But since this is becoming proper public API now, we should make sure all the combinations of parameters work correctly.